src/fpm/manifest/profiles.f90

```diff
@@ -0,0 +1,835 @@
+                end associate
+              end do
+            end if
+          end do
+        end if
+
+        profiles(profindex) = new_profile(profile_name, compiler_name, os_type, &
+                 & flags, c_flags, link_time_flags, file_scope_flags)
+        profindex = profindex + 1
+      end subroutine get_flags
+
+      !> Traverse operating system tables
+      subroutine traverse_oss(profile_name, compiler_name, os_list, table, error, profiles_size, profiles, profindex)
+        
+        !> Name of profile
+        character(len=:), allocatable, intent(in) :: profile_name
+
+        !> Name of compiler
+        character(len=:), allocatable, intent(in) :: compiler_name
+
+        !> List of OSs in table with profile name and compiler name given
+        type(toml_key), allocatable, intent(in) :: os_list(:)
+
+        !> Table containing OS tables
+        type(toml_table), pointer, intent(in) :: table
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        !> Number of profiles in list of profiles
+        integer, intent(inout), optional :: profiles_size
+
+        !> List of profiles
+        type(profile_config_t), allocatable, intent(inout), optional :: profiles(:)
+
+        !> Index in the list of profiles
+        integer, intent(inout), optional :: profindex
+        
+        type(toml_key), allocatable :: key_list(:)
+        character(len=:), allocatable :: os_name, l_os_name
+        type(toml_table), pointer :: os_node
+        character(len=:), allocatable :: flags
+        integer :: ios, stat, os_type
+        logical :: is_valid, key_val_added, is_key_val
+
+        if (size(os_list)<1) return
+        key_val_added = .false.
+        do ios = 1, size(os_list)
+          os_name = os_list(ios)%key
+          call validate_os_name(os_name, is_valid)
+          if (is_valid) then
+            if (present(profiles_size)) then
+              profiles_size = profiles_size + 1
+            else
+              if (.not.(present(profiles).and.present(profindex))) then
+                call fatal_error(error, "Both profiles and profindex have to be present")
+                return
+              end if
+              call get_value(table, os_name, os_node, stat=stat)
+              if (stat /= toml_stat%success) then
+                call syntax_error(error, "os "//os_name//" has to be a table")
+                return
+              end if
+              call match_os_type(os_name, os_type)
+              call os_node%get_keys(key_list)
+              call get_flags(profile_name, compiler_name, os_type, key_list, os_node, profiles, profindex, error)
+              if (allocated(error)) return
+            end if
+          else
+            ! Not lowercase OS name
+            l_os_name = lower(os_name)
+            call validate_os_name(l_os_name, is_valid)
+            if (is_valid) then
+              call fpm_stop(1,'*traverse_oss*:Error: Invalid OS name.')
```

Comment:
We can't stop while reading the manifest, instead create an error and propagate up.